File: Competitive Coding/C/QuickSort.c
#include <stdio.h>
void swap(int *a,int *b)
{
    int temp=*a;
    *a=*b;
    *b=temp;
}
int partition(int *ar,int p,int r)
{
    int i=p-1,j=p,x=ar[r];
    while(j<r)
    {
        if(ar[j]>x)
        j++;
        else if(ar[j]<x)
        {
            i++;
            swap(&ar[i],&ar[j]);
            j++;
        }
    }
    swap(&ar[r],&ar[i+1]);
    return i+1;
}
void quicksort(int *ar,int p,int r)
{
    if(p<r)
    {
    int q=partition(ar,p,r);
    quicksort(ar,p,q-1);
    quicksort(ar,q+1,r);
    }
}
int main()
{
int size,i,j;
scanf("%d",&size);
int ar[size];
for(i=0;i<size;i++)
scanf("%d",&ar[i]);
quicksort(ar,0,size-1);
for(i=0;i<size;i++)
printf("%d ",ar[i]);
    return 0;
}

File: Competitive Coding/Java/Anchal Goyal/quickSort.java
import java.util.Scanner;
// Other imports go here
// Do NOT change the class name
class quickSort
{
    public static int partition(int a[],int start,int end)
    {
      int pivot=end;
      int i,j=end,count=0;
      for(i=0;i<=end;i++)
      {
        if(a[i]<=a[pivot])
        {
            count++;
        }
      }
     int temp=a[count-1];
      a[count-1]=a[pivot];
      a[pivot]=temp;
      pivot=count-1;

    i=0;
    while(i<pivot && j>pivot)
    {
        if(a[i]<=a[pivot])
        {
            i++;
        }
        else if(a[j]>a[pivot])
        {
            j--;
        }
        else
        {
            swap(a,i,j);
            i++;
            j--;
        }
    }
    return pivot;
    }
public static void swap(int a[],int x,int y)
{
    int temp;
    temp=a[x];
    a[x]=a[y];
    a[y]=temp;
}
    public static void sort(int a[],int start,int end)
    {
        if(start<end)
        {
            int x=partition(a,start,end);
            sort(a,start,x-1);
            sort(a,x+1,end);
        }
    }
    public static void main(String[] args)
    {
           // Write your code here
        Scanner s=new Scanner(System.in);
       System.out.print("Enter number of elements");
        int n=s.nextInt();
        int arr[]=new int[n];
        for(int i=0;i<n;i++)
        {
            arr[i]=s.nextInt();
        }
        sort(arr,0,n-1);
        for(int i=0;i<n;i++)
        {
            System.out.print(arr[i]+" ");
        }
    }
    }